src/native/profiler.cc

```diff
@@ -344,15 +345,24 @@ Profiler::runAgentThread(jvmtiEnv *jvmti_env, JNIEnv *jni_env, void *args) {
       jint end_line; //exclusive
       jint line = -1;
       std::vector<std::pair<jint, jint>> location_ranges;
+
+      bool select_last_line = true;
       for (int i = 1; i < num_entries; i++) {
         if (line == -1
             && entries[i].start_location > exp_frame.lineno) {
           line = entries[i - 1].line_number;
           current_experiment.lineno = line;
+          select_last_line = false;
           break;
         }
       }
-      for (int i = 1; i < num_entries; i++) {
+
+      if (select_last_line && num_entries > 0) {
+        line = entries[num_entries - 1].line_number;
+        current_experiment.lineno = line;
+      }
+
+      for (int i = 0; i < num_entries; i++) {
```

Comment:
Wow, I'm surprised the bug of not starting at index `0` has been here the whole time.